### PR TITLE
fix: adds missing type for response bodies

### PR DESCRIPTION
**Pull request by jcsackett**
Fixes: https://github.com/snyk/sweater-comb/issues/257

### CODE REVIEW

File: src/rulesets/rest/2022-05-25/json-api-rules/resource-object-rules.ts [hunk 1]
```diff
@@ -17,11 +17,13 @@ const responseDataForPatch = new ResponseBodyRule({
   rule: (responseAssertions) => {
     responseAssertions.body.added.matches({
       schema: {
+        type: 'object',
```
Comment (cmars) [inline src/rulesets/rest/2022-05-25/json-api-rules/resource-object-rules.ts:20]: This is slightly better than what we had before, but what we had wasn't very good. This should be asserting a [valid JSON API PATCH response](https://jsonapi.org/format/#crud-updating-responses) for a 200, similar to how it's done below for GET and POST responses (see `dataProperty` below).

Comment (cmars) [reply src/rulesets/rest/2022-05-25/json-api-rules/resource-object-rules.ts:20]: I wonder if we should refactor a bit -- a rule requiring a JSON API resource response could use a match function that looks at method, response code and content type, to consolidate the decision logic on when a response should contain a resource (200 or 201 for a POST, 200 for GET and PATCH, etc.)

**Review by cmars — COMMENTED** [review]
Still needs tests.

To really require `type:` for all schemas, it will need to be done recursively, similar to what was done in #240, to ensure all nested schemas in properties have `type:` declared as well.

That should be done in a separate rule (though a schema-walker might be refactored out of #240).

**Comment by jcsackett** [comment]
To make sure we're on the same page, we're good with this fix for the specific issue but you're noting a separate issue that we don't have full validation of type for all schemas. 

Assuming that's a correct understanding I'll land this and create an issue for the recursive aspect.

**Pull request merged**